File: 유진/4월1주차/[PGS]덧칠하기.js
[
  function solution(n, m, section) {
    var answer = 0;
    var pos = [];

    for (let i = 1; i <= n; i++) {
      if (i == section[0]) {
        pos[i] = 0;
        section.shift();
      } else {
        pos[i] = 1;
      }
    }

    let pp = 0;
    let count = 0;

    while (pos.includes(0)) {
      if (pos[pp] == 0) {
        for (let a = pp; a < m + pp; a++) {
          pos[a] = 1;
        }
        count++;
      }
      pp++;
    }

    answer = count;
    return answer;
  }
];

File: 성은/[PGS]덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        for(int i=0;i<section.length;i++){
            answer++;
            int a=0; //뛰어넘을 값 체크
            for(int j=i+1;j<section.length;j++){
                if(section[j]<=section[i]+m-1) a++;
            }
            i+=a;
        }
        return answer;
    }
}